AppCenterDistribute/AppCenterDistribute/MSDistribute.m

```diff
@@ -589,20 +590,34 @@ - (void)openURLInAuthenticationSessionWith:(NSURL *)url API_AVAILABLE(ios(11)) {
     if (callbackUrl) {
       [strongSelf openURL:callbackUrl];
     }
+    [[MSUtility sharedApp] endBackgroundTask:backgroundAuthSessionTask];
   };
   SFAuthenticationSession *session = [[SFAuthenticationSession alloc] initWithURL:url
                                                                 callbackURLScheme:callbackUrlScheme
                                                                 completionHandler:authCompletionBlock];
 
+  if (self.authenticationSession != nil) {
+
+    // Calling 'start' on an existing session crashes the application - cancel session.
+    [(SFAuthenticationSession *)self.authenticationSession cancel];
```

Comment:
Why do we still use cast instead just storing the correct type?

Reply:
SFAuthenticationSession is only used in ios 11 and above

Reply:
Not resolved. The entire method only for iOS 11 (even if not it doesn't change this point)

Reply:
Updated